neo/Consensus/ConsensusService.cs

```diff
@@ -293,7 +296,7 @@ private void OnRecoveryMessageReceived(ConsensusPayload payload, RecoveryMessage
                         ReverifyAndProcessPayload(changeViewPayload);
                 }
                 if (message.ViewNumber != context.ViewNumber) return;
-                if (!context.CommitSent())
+                if (!context.ViewChanging() && !context.CommitSent())
```

Comment:
You should allow receiving `PrepareRequest` and `PrepareResponse` even when `ViewChanging() == true`. Because `ViewChanging()` indicates that I want to change views, it may not be able to change successfully.

Reply:
See the comments below if it accepts prepare requests and prepare responses in the current view after it has sent a `ChangeView` message it can end up sending the other nodes to the next view, where they will never be able to receive the recovery message from this view with the commits, since it will be a lower view number; thus the network will be stalled.

Reply:
If the view has not been changed, then we should allow consensus on the current view. Otherwise, we may not be able to reach a consensus or change views.

Reply:
If the view is changing for a node (then the view has to be considered to be changed from it's perspective), because it has already sent a change view message that can be used to allow others to accept a change view, so it cannot commit in the current view. If it did commit in the current view after sending a change view, then the others can all accept moving to the later view, where they won't be able to reach consensus.

If more than `F` nodes have committed, then this code does allow receiving `PrepareRequest` and `PrepareResponse` in the current view effectively to obtain consensus by treating the `ViewChanging` flag as not changing. Maybe you missed this line that is changed in this PR:

https://github.com/neo-project/neo/blob/470b35c63705c2770fe40346c90b6e2f1d2d148d/neo/Consensus/Helper.cs#L32


If less than `F` nodes have been committed then they will still be able to obtain consensus in a higher view. So this code seems to address your concern and nodes will be able to reach consensus.